src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -655,7 +665,20 @@ def _fix_pandas_df_integer(
             and column_metadata.scale == 0
             and not str(pandas_dtype).startswith("int")
         ):
-            pd_df[pandas_col_name] = pandas.to_numeric(
-                pd_df[pandas_col_name], downcast="integer"
-            )
+            # pandas.to_numeric raises "RuntimeWarning: invalid value encountered in cast"
+            # when it tried to downcast and loses precision. In this case, we try to convert to
+            # int64 using astype() and fallback to to_numeric if we were unsuccessful.
+            with warnings.catch_warnings(record=True) as warning_list:
+                pd_col_with_numeric_downcast = pandas.to_numeric(
+                    pd_df[pandas_col_name], downcast="integer"
+                )
+
+            if not warning_list:
```

Comment:
Is it possible to check the warning_list content?

Reply:
it is possible. what kind of logic would you suggest be better here? `if len(warning_list) == 1 and warning_list[0].message.startswith("RuntimeWarning")` ?

Reply:
Does the warning have a specific text? I no this is good. If yes I suggest you check if that text is in the warning.

Reply:
it does: `RuntimeWarning('invalid value encountered in cast')`

Reply:
Yeah checking this text will be better just in case there is another different warning.